src/commcare_cloud/ansible/setup_pg_logical_replication.yml

```diff
@@ -0,0 +1,144 @@
+      postgresql_ext:
+        name: pglogical
+        db: "{{item.master_db_name}}"
+        port: "{{ postgresql_port }}"
+      with_items: "{{ postgresql_dbs.logical }}"
+      when: item.master_host == inventory_hostname
+
+    - name: Create pglogical extension (subscribers)
+      become: yes
+      become_user: postgres
+      vars:
+        ansible_ssh_pipelining: true
+      postgresql_ext:
+        name: pglogical
+        db: "{{item.name}}"
+        port: "{{ postgresql_port }}"
+      with_items: "{{ postgresql_dbs.logical }}"
+      when: item.host == inventory_hostname
+
+# Set up provider nodes
+- hosts:
+    - shard_dbs
+  become: yes
+  become_user: postgres
+  vars_files:
+    - roles/postgresql_base/defaults/main.yml
+  vars:
+    - form_processor_replication_tables:
+        - table_name: form_processor_xforminstancesql
+          hash_fn: "hash_string(form_id, ''siphash24'')"
+        - table_name: form_processor_xformoperationsql
+          hash_fn: "hash_string(form_id, ''siphash24'')"
+        - table_name: form_processor_xformattachmentsql
+          hash_fn: "hash_string(form_id, ''siphash24'')"
+        - table_name: form_processor_commcarecasesql
+          hash_fn: "hash_string(case_id, ''siphash24'')"
+        - table_name: form_processor_commcarecaseindexsql
+          hash_fn: "hash_string(case_id, ''siphash24'')"
+        - table_name: form_processor_caseattachmentsql
+          hash_fn: "hash_string(case_id, ''siphash24'')"
+        - table_name: form_processor_casetransaction
+          hash_fn: "hash_string(case_id, ''siphash24'')"
+        - table_name: form_processor_ledgervalue
+          hash_fn: "hash_string(case_id, ''siphash24'')"
+        - table_name: form_processor_ledgertransaction
+          hash_fn: "hash_string(case_id, ''siphash24'')"
+        - table_name: scheduling_casealertscheduleinstance
+          hash_fn: "hash_string(case_id, ''siphash24'')"
+        - table_name: scheduling_casetimedscheduleinstance
+          hash_fn: "hash_string(case_id, ''siphash24'')"
+        - table_name: scheduling_alertscheduleinstance
+          hash_fn: "hash_string(decode(replace(schedule_instance_id::text, ''-'', ''''), ''hex''), ''siphash24'')"
+        - table_name: scheduling_timedscheduleinstance
+          hash_fn: "hash_string(decode(replace(schedule_instance_id::text, ''-'', ''''), ''hex''), ''siphash24'')"
+        - table_name: blobs_blobmeta
+          hash_fn: "hash_string(parent_id, ''siphash24'')"
+        - table_name: blobs_deletedblobmeta
+          hash_fn: "hash_string(parent_id, ''siphash24'')"
+  tasks:
+    - name: Create node
+      postgresql_query:
+        db: '{{ item.master_db_name }}'
+        query: "SELECT pglogical.create_node(node_name := '{{ item.master_db_name }}_provider', dsn := 'host={{ item.master_host }} port=5432 dbname={{ item.master_db_name }} user={{ postgres_users.replication.username}} password={{ postgres_users.replication.password }}')"
```

Comment:
I always received postgresql syntax errors when trying to split this query between multiple lines. I eventually gave up and made all of these large lines.